src/sentry/models/event.py

```diff
@@ -496,13 +496,19 @@ def tags(self):
     def get_minimal_user(self):
         from sentry.interfaces.user import User
 
+        if all(key in self.snuba_data for key in ["user_id", "email", "username", "ip_address"]):
+            user_id = self.snuba_data["user_id"]
+            email = self.snuba_data["email"]
+            username = self.snuba_data["username"]
+            ip_address = self.snuba_data["ip_address"]
```

Comment:
Would you ever want to partially read the user data from snuba and get the missing attributes from nodestore?

Reply:
Not that I'm aware of. The rationale for looking in `snuba_data` first is to avoid the nodestore load if it's not necessary. But if we need to get even one field from nodestore then we will have loaded the entire object and all the fields will be there.